Add "Deposit Every" field to "Create Recurring Deposit Product" page

AkashBhave opened the pull request:
I added the "Deposit every" field to the "Create Recurring Deposit Product" page. This page is located at Admins > Products > Recurring Deposit Products > Create Recurring Deposit Product > Settings. I also added the label to the individual page for each Recurring Deposit Products. 
![screenshot1](https://user-images.githubusercontent.com/16819026/33641218-948b8b84-da02-11e7-9f9d-6fa00d3f7869.PNG)
![screenshot2](https://user-images.githubusercontent.com/16819026/33641219-949ca856-da02-11e7-9f2c-197c239aa740.PNG)
Part of a task for Google Code-in 2017: https://codein.withgoogle.com/tasks/5947057493770240/

mbj36 commented:
Thanks for the PR, It will be reviewed Shortly!

This message was posted by [GitMate.io](https://gitmate.io) with timestamp signature `Thanks for the PR, It will be reviewed Shortly!:1eMPOL:Jrpke-i6XAdySgThiOb443UcGgE`

mbj36 commented:
@gkrishnan724 Can you please review this ?

gkrishnan724 commented on app/views/products/createrecurringdepositproduct.html:
```diff
@@ -214,6 +214,26 @@
+                            <select id="depositEveryFrequencyType" ng-model="formData.depositEveryFrequencyType"
+                                    class="form-control"
+                                    ng-options="type.id as type.value for type in product.depositEveryFrequencyTypeOptions"
+                                    ng-change="depositproduct.depositEveryFrequencyOption.value = formValue(product.depositEveryFrequencyTypeOptions,formData.depositEveryFrequencyType)"
```
`depositproduct.depositEveryFrequencyOption.value` Should be changed to `depositproduct.depositEveryFrequencyOption`.  The preview step should also display the field input. Take a look at the code in the preview step, you will understand how to make this input visible over there. It is similar to the way we display it in `viewreccuringdepositproduct.html`

AkashBhave commented:
@gkrishnan724 I just added the changes that you told me to add:

- Added field to `app/views/reccurring/newapplication.html`
![screenshot3](https://user-images.githubusercontent.com/16819026/33692714-ab33607c-dabc-11e7-91e2-eb9c87658e9e.PNG)

- Added field to "Preview" section of `app/views/products/createrecurringdepositproduct.html`

I also based the field off of `minDepositTerm` instead of `lockinPeriod` because the field options were not showing properly previously.

However, I am not sure if the changes are applied to the back-end, so not everything might work.

gkrishnan724 commented on app/views/products/createrecurringdepositproduct.html:
```diff
@@ -850,6 +869,11 @@ <h3>{{depositproduct.name}}</h3>
                         </tr>-->
+                        <tr>
+                            <td>{{'label.heading.depositEveryFrequency' | translate}}</td>
+                            <td>{{depositproduct.depositEveryFrequency}} &nbsp; {{depositproductdepositEveryFrequencyType.value}}
```
You forgot a `.` here it is `depositproduct.depositEveryFrequencyType.value`

AkashBhave commented:
What a silly error! Fixed and pushed as one commit to the PR.

gkrishnan724 commented:
PR will be merged after the backend has been Implemented. CC @nazeer1100126
